### PR TITLE
feat: [IOCOM-172,IOCOM-173,IOCOM-179] Display remote precondition before opening the message details

### PR DESCRIPTION
**Pull request by adelloste**
## Short description
This PR handles message precondition. A bottom sheet is going to be shown with a remote content if the message has precondition enabled (`hasPrecondition`), otherwise browse to the message details.

https://user-images.githubusercontent.com/29163287/234013474-46bad309-69dd-46f2-a5c1-e44cedb6fe66.mp4

## List of changes proposed in this pull request
- Added async action to get `ThirdPartyMessagePrecondition`
- Added standard action to clear `ThirdPartyMessagePrecondition`
- Added `thirdPartyMessagePreconditionById` reducer 
- Added saga
- Removed locales
- Merged `PnOpenConfirmationBottomSheet.tsx` into `useMessageOpening.tsx`

## How to test
- Checkout on [this branch](https://github.com/pagopa/io-dev-api-server/tree/IOCOM-171-remote-bottom-sheet-message)
- Enable `includePn` in the [dev server config](https://github.com/pagopa/io-dev-api-server/blob/IOCOM-171-remote-bottom-sheet-message/src/config.ts#L131)
- Increment `pnCount` in the [dev server config](https://github.com/pagopa/io-dev-api-server/blob/IOCOM-171-remote-bottom-sheet-message/src/config.ts#L78)
- Perform a login
- Open message

### CODE REVIEW

File: ts/store/actions/messages.ts
Comment (Vangaorth) [inline ts/store/actions/messages.ts:143]: Maybe "_load_" is not the right term for the action-triggered behaviour? After the `.request` is called, we trigger a web request so maybe a _getThirdPartyMessagePrecondition_ is better suited?

File: ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts
Comment (Vangaorth) [inline ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts:1]: This reducer has "_byId_" in its name but it is not really sorting/handling multiple preconditions by ids, it handles a single object. Maybe we can drop the "_byId_"?

File: ts/store/reducers/entities/messages/index.ts
Comment (Vangaorth) [inline ts/store/reducers/entities/messages/index.ts:34]: See comment on `ts/store/reducers/entities/messages/messagesById.ts`

File: ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts
Comment (Vangaorth) [inline ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts:55]: This selector should not need to be cached using `createSelector`, since it is just reading a property from memory and returns it without creating a new instance. Can we remove the caching?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts
Comment (adelloste) [reply ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts:1]: Renamed `thirdPartyMessagePreconditionById` to `messagePrecondition`.

File: ts/store/reducers/entities/messages/index.ts
Comment (adelloste) [reply ts/store/reducers/entities/messages/index.ts]: Removed all references to `thirdPartyMessagePreconditionById`

File: ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts
Comment (adelloste) [reply ts/store/reducers/entities/messages/thirdPartyMessagePreconditionById.ts]: Fixed https://github.com/pagopa/io-app/pull/4565/commits/8874819a02a6fa9b3576398a7621b1368aa02313

File: package.json
Comment (hevelius) [inline package.json:4]: why not use tag release instead commit?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: package.json
Comment (hevelius) [reply package.json:4]: I suppose it's not already deployed. :-D

Comment (adelloste) [reply package.json:4]: Exactly, we are waiting for the release :)

**Review by hevelius — APPROVED** [review]
LGTM. Tested on iOS simulator. It would be better before merge use a release tag instead a commit for new io-backend specs.

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**
